polylang/polylang
# New filter for `PLL_Language` URLs instead of `site_url()`.

## What?
Fixes #1296 

## Why?
See https://github.com/polylang/polylang/issues/1296#issuecomment-1572107562
`site_url()` adds an extra `/subdir/`.

## How?
⚠️ ~~Introduce `pll_language_url` to filter language URL (especially for domains).~~ changed the way to fix the issue, see bellow.
- Add `pll_language_flag_url` filter hooked by `PLL_Links_Abstract_Domain::site_url()`
- Add `pll_home_url` filter hooked by `PLL_Links_Model::set_language_home_url()`
- Add `pll_search_url` filter hooked by `PLL_Links_Model::set_language_search_url()`

## Test instruction
Add the following code to your `pll-config.php` file:
```php
<?php

function pll_log_lang_urls() {
    $languages = PLL()->model->get_languages_list();

    if ( ! defined( 'PLL_CACHE_LANGUAGES' ) ) {
        define( 'PLL_CACHE_LANGUAGES', false );
    }

    foreach ( $languages as $language ) {
        error_log( "{$language->name} Flag URL: {$language->get_display_flag_url()}" );
        error_log( "{$language->name} Home URL: {$language->get_home_url()}" );
        error_log( "{$language->name} Search URL: {$language->get_search_url()}" );
    }
}
add_action( 'pll_init', 'pll_log_lang_urls' );
```

Clean the transient and visit any page of your site. Check your `debug.log`, the logged URL should be correct.

*Note: Allow `.jpeg` extension for custom flags. And clean `PLL_Domain_UnitTestCase`*
